### DISCUSSION TITLE
Scheduled tasks for non http requests

### DISCUSSION THREAD

**Discussion by BennyG93** [coollabsio/coolify#4231] [Q&A]
Hello, can the Schedules Tasks feature be used for non-http triggered tasks? I have a project that is a monorepo, it has a fastapi application and a bunch of python scripts for various tasks. I don't really want to include these tasks in the api if I don't have to.

In my project repo I have 2 main directories:
- `app/`  where the fastapi app lives
- `functions/` where my python scripts live

The application itself is configured to build and deploy using nixpack and the fastapi start command, in my case `fastapi run app/main.py`. So there is no docker container environment.

When I try to configure a scheduled task to run something like `python function/update_something.py`, I get some python module errors like:
```
ModuleNotFoundError: No module named 'requests'
```

My take aways from this are:
- Python is installed
- My functions files are accessible
- But the environment is not configured

So I tried to be smart and changed the command to `pip install -r requirements.txt && python function/update_something.py`. Thinking I'll just install all my modules and then run the command.... nope:
```
sh: 1: pip: not found
```

One more attempt at being smart: `python -m pip install -r requirements.txt && python function/update_something.py`
```
/nix/var/nix/profiles/default/bin/python: No module named pip
```

So, is there a way to create Scheduled Tasks for executable python scripts instead of wrapping them in an api endpoint?

**Comment by developeralgo** [comment]
I came across the same situation where I need to run scripts but not in an api fashion. Have no idea how to proceed.
